web_src/less/_repository.less

```diff
@@ -1301,21 +1281,63 @@
           padding: 5px 10px;
         }
       }
+
+      @media @mediaSm {
+        flex-direction: column;
+
+        > * + * {
+          margin-top: .5em;
+        }
+      }
+
+      @media (max-width: 480px) {
+        form {
+          flex-direction: column;
+          align-items: stretch !important;
+
+          > * + * {
+            margin-top: .5em;
+          }
+        }
+      }
+    }
+  }
+
+  .commits-table-left {
+    white-space: nowrap;
+    min-width: 0;
+
+    span {
+      overflow: hidden;
+      text-overflow: ellipsis;
     }
   }
 
   #commits-table {
+    @media (max-width: 480px) {
+      .ui.avatar,
+      th.sha,
+      td.sha {
+        display: none;
+      }
+    }
+
     thead {
       th:first-of-type {
         padding-left: 15px;
       }
 
-      .sha {
-        &td {
-          text-align: center;
+      .sha.mono {
+        font-size: 1em !important;
+        width: calc(9ch + 82px); // 60px + 11px * 2
```

Comment:
Do we really need explicit width on this? I prefer to avoid that.

Reply:
This is necessary since the table is fixed, i. e. the width of the columns are have predefined starting values (can be incremented but not decreased). I basically copied the logic from the commits table to the repo file table for consistency.

Reply:
So essentially you are stretching the column with this `width`? Can we be sure it will not cut off for example if we add more content in the hash label?

Can you maybe convert to `min-width`?

Reply:
> So essentially you are stretching the column with this `width`? Can we be sure it will not cut off for example if we add more content in the hash label?
> 

I'm pretty sure it will cut off if more content is added.

> Can you maybe convert to `min-width`?

It won't work. If I recall correctly, when the width of a fixed table decreases, only explicit `width` (either as percentage in explicit numbers) could have an effect on column widths. No actual content width would be taken in to consideration. It's common practice though and ensures the fastest layout possible for large tables.

Reply:
To make it clear again: the commits table has always been fixed, and there wasn't any measure taken to ensure that the SHA label would fit (the Fomantic UI classes define the percentage width of the columns, just like the flex-based grid (but does not support that many layout strategies). It basically gives up at lower resolution and wraps the table in three rows - that design may find its use elsewhere, but the UE is awful in this case. I thus use media queries to hide the less-important elements explicitly, but otherwise keep the design same as before (it has the advantage of always getting the `overflow-text: ellipsis` right for commit messages embedded in flexboxes, so I also ported that to the repo file table by using a complex table header with a zero-height borderless top row, used just to set the width of the columns - I'm yet to find anything more elegant than that based on what's already there).

Ideally, we could copy GitHub's idea of using a bunch of nested flexboxes (with helper classes based on media queries) given their versatility, but those are pretty hard to tame...

Reply:
Also, thanks for all the reviews. I've read all of them and have them on my checklist. Have other things to do...